src/lib/Key.js

```diff
@@ -91,6 +161,124 @@ class Key {
+            'spki',
+            this.rsaKeyPair.publicKey,
+            { name: 'RSA-OAEP', hash: 'SHA-256' },
+            true, // Allow extraction
+            ['encrypt'],
+        );
+    }
+
+    /**
+     * @param {EncryptionKeyParams} keyParams
+     * @returns {Promise<RsaKeyPairExport>}
+     */
+    async _computeRsaKeyPair(keyParams) {
+        const iframe = document.createElement('iframe');
+        iframe.classList.add('rsa-sandboxed-iframe'); // Styles in common.css hide this class
+        iframe.setAttribute('sandbox', 'allow-scripts');
+        iframe.src = '../../lib/rsa/sandboxed/RSAKeysIframe.html'; // Relative path from a request URL
+        /** @type {Promise<void>} */
+        const loadPromise = new Promise(resolve => iframe.addEventListener('load', () => resolve()));
+        document.body.appendChild(iframe);
+        await loadPromise;
+
+        if (!iframe.contentWindow) {
+            throw new Error('Could not load sandboxed RSA iframe');
+        }
+
+        // Extend 32-byte secret into 1024-byte seed as bytestring
+        /** @type {Uint8Array} */
+        let seed;
+        switch (keyParams.kdf) {
+            case 'PBKDF2-SHA512':
+                seed = Nimiq.CryptoUtils.computePBKDF2sha512(
+                    this.secret.serialize(),
+                    this._defaultAddress.serialize(),
+                    keyParams.iterations,
+                    1024, // Output size (required)
+                );
+                break;
+            default:
+                throw new Error(`Unsupported KDF function: ${keyParams.kdf}`);
+        }
+
+        // Send computation command to iframe
+        iframe.contentWindow.postMessage({
+            command: 'generateKey',
+            seed: Nimiq.BufferUtils.toUtf8(seed), // seed is a bytestring
+            keySize: keyParams.keySize,
+        }, '*');
+
+        /** @type {(keyPair: RsaKeyPairExport) => void} */
+        let resolver;
+        /** @type {Promise<RsaKeyPairExport>} */
+        const resultPromise = new Promise(resolve => {
+            resolver = resolve;
+        });
+
+        /**
+         * @param {MessageEvent} event
+         */
+        function onMessage(event) {
+            if (event.source === event.target) {
+                // console.log("Ignored same-window event:", event);
+                return;
+            }
+
+            /** @type {{privateKey: ArrayBuffer, publicKey: ArrayBuffer}} */
+            const data = event.data;
+            if (!('privateKey' in data) || !('publicKey' in data)) return;
+
+            window.removeEventListener('message', onMessage);
+
+            resolver({
+                privateKey: new Uint8Array(data.privateKey),
+                publicKey: new Uint8Array(data.publicKey),
+                keyParams,
+            });
+        }
+
+        // Listen for result from iframe
+        window.addEventListener('message', onMessage);
+
+        return resultPromise;
```

Comment:
I think the Promise reads nicer like this:
```suggestion
        return new Promise(resolve => {
            /**
             * @param {MessageEvent} event
             */
            function onMessage(event) {
                if (event.source === event.target) {
                    // console.log("Ignored same-window event:", event);
                    return;
                }

                /** @type {{privateKey: ArrayBuffer, publicKey: ArrayBuffer}} */
                const data = event.data;
                if (!('privateKey' in data) || !('publicKey' in data)) return;

                window.removeEventListener('message', onMessage);

                resolve({
                    privateKey: new Uint8Array(data.privateKey),
                    publicKey: new Uint8Array(data.publicKey),
                    keyParams,
                });
            }

            // Listen for result from iframe
            window.addEventListener('message', onMessage);
        });
```

Reply:
Resolved in #524